subsys/shell/shell.c

```diff
@@ -622,7 +606,8 @@ static int execute(const struct shell *shell)
 
 		if (IS_ENABLED(CONFIG_SHELL_HELP) &&
 		    (!strcmp(argv[cmd_lvl], "-h") ||
-		     !strcmp(argv[cmd_lvl], "--help"))) {
+		     !strcmp(argv[cmd_lvl], "--help")) &&
+		    (cmd_lvl > 0)) {
```

Comment:
Moving this up 2 lines would improve readability and execution time.

Reply:
done